src/libraries/System.Private.CoreLib/src/System/SpanSplitEnumerator.T.cs

```diff
@@ -76,7 +76,22 @@ public bool MoveNext()
             ReadOnlySpan<T> slice = _buffer.Slice(_startNext);
             _startCurrent = _startNext;
 
-            int separatorIndex = _splitOnSingleToken ? slice.IndexOf(_separator) : slice.IndexOf(_separators);
+            int separatorIndex;
```

Comment:
Whats the motivation for this change? To improve runtime perf of Split(null) to make it not O(n^2)?